### PR TITLE
single template for each resource type

### PR DESCRIPTION
**Pull request by deepak-ms**
## Description

creates single template for each resource type (from having one template per resource per message type)

## Related issues

Addresses [issue #].

## Testing

Tested Manually

### CODE REVIEW

File: src/lib/handlebars-converter/handlebars-converter.spec.js [hunk 1]
```diff
@@ -17,8 +17,8 @@ describe('HandlebarsConverter', function () {
     var handlerbarExistingInstance = HandlebarsConverter.instance(false, path.join(__dirname, '../../templates'));
 
     var templatesMap = {};
-    templatesMap['Resources/ADT_A01/Patient.hbs'] = '{"key1":"value1"}';
-    templatesMap['Resources/ADT_A01/Encounter.hbs'] = '{"key2":"value2"}';
+    templatesMap['Resources/Patient.hbs'] = '{"key1":"value1"}';
```
Comment (CaitlinV39) [inline src/lib/handlebars-converter/handlebars-converter.spec.js]: What happens to the code if they remove the patient resource template for some reason?

File: src/routes.spec.js [hunk 2]
```diff
@@ -1540,7 +1540,7 @@ describe('POST /api/convert/hl7/:template (with stored template)', function () {
 
     it('should return 200 OK with valid message and existing template in a subdirectory', function (done) {
         supertest(app)
-            .post("/api/convert/hl7/Resources/ADT_A01/Patient.hbs")
+            .post("/api/convert/hl7/Resources/Patient.hbs")
```
Comment (CaitlinV39) [inline src/routes.spec.js]: Same question on this. What happens to the overall code if they Resources/Patient template is removed for some reason?

File: src/lib/handlebars-converter/handlebars-converter.spec.js (hunk 1 above)
Comment (deepak-ms) [reply src/lib/handlebars-converter/handlebars-converter.spec.js]: it's just a test code. Users can update their repo if the related template is removed and they still want to maintain the test success rate in their repo.

File: src/routes.spec.js (hunk 2 above)
Comment (deepak-ms) [reply src/routes.spec.js]: same comment as above.

File: src/templates/Resources/PractitionerRole.hbs [hunk 3]
```diff
@@ -4,31 +4,40 @@
         "resourceType": "PractitionerRole",
         "id":"{{ID}}",
         {{>DataType/XCNPractitionerRole.hbs XCN=PD1-4}},
-        {{>DataType/XCNPractitionerRole.hbs XCN=PRT-5}},
```
Comment (CaitlinV39) [inline src/templates/Resources/PractitionerRole.hbs]: I think that PRT-5 does have an identifier component?

File: src/templates/Resources/PractitionerRole.hbs [hunk 4]
```diff
@@ -4,31 +4,40 @@
...
+        {{>DataType/XCNPractitionerRole.hbs XCN=PRT-5}},
+        "specialty":
         [
-            {
-                {{>DataType/PLN.hbs PLN=PRT-24}},
-            },
+            { {{>DataType/CWECodeableConcept.hbs CWE=PRT-6}} },
         ],
-        {{>DataType/XCNPractitionerRole.hbs XCN=OBX-25}},
+        {{>DataType/XCNPractitionerRole.hbs XCN=ORC-12}},
```
Comment (CaitlinV39) [inline src/templates/Resources/PractitionerRole.hbs]: I dont think ORC12 has practitioner role, just practitioner

File: src/templates/Resources/Account.hbs [hunk 5]
```diff
@@ -8,10 +8,10 @@
             { {{>DataType/CX.hbs CX=PID-18}} },
         ],
         "subject":
-        {
+        [
```
[the remark this answers is not in the document]
Comment (deepak-ms) [reply src/templates/Resources/Account.hbs]: Please open a new issue for Patient ID. We are not handling the non-literal assignments in FHIR attribute field from spreadsheets right now. Commas are anyway ignored in post-processing.

File: src/templates/Resources/PractitionerRole.hbs (hunk 3 above)
Comment (deepak-ms) [reply src/templates/Resources/PractitionerRole.hbs]: this line still exists on right hand side (moved a little lower)

File: src/templates/Resources/Account.hbs (hunk 5 above)
[the remark this answers is not in the document]
Comment (CaitlinV39) [reply src/templates/Resources/Account.hbs]: Sounds good! Added a bug.

File: src/templates/Resources/PractitionerRole.hbs (hunk 3 above)
Comment (CaitlinV39) [reply src/templates/Resources/PractitionerRole.hbs]: Sorry! I caught a couple of these but not all of them.

File: src/templates/Resources/PractitionerRole.hbs (hunk 4 above)
Comment (deepak-ms) [reply src/templates/Resources/PractitionerRole.hbs]: Probably its coming from this spreadsheet : https://docs.google.com/spreadsheets/d/11VBF5TIRJTfRwkRIK1XH-k40dZqCd3g3a0sBY-zJU4Q/edit#gid=0

File: src/templates/Resources/Specimen.hbs [hunk 6]
```diff
@@ -3,8 +3,9 @@
     "resource":{
         "resourceType": "Specimen",
         "id":"{{ID}}",
-        "receivedTime":"{{convertDateTimeStringToUTC OBR-14}}",
-        "receivedTime":"{{convertDateTimeStringToUTC SPM-18}}",
+        "receivedTime":"{{formatAsDateTime OBR-14}}",
+        "receivedTime":"{{formatAsDateTime SPM-18}}",
+        "receivedTime":"{{formatAsDateTime OBR-13}}",
```
[the remark this answers is not in the document]
Comment (deepak-ms) [reply src/templates/Resources/Specimen.hbs]: coming from https://docs.google.com/spreadsheets/d/1kd4z6vi8QtN2BqmaAWFJ6tzCnUQzzpybJAVCeqKkzCg/edit#gid=0

File: src/templates/Resources/PractitionerRole.hbs (hunk 4 above)
Comment (CaitlinV39) [reply src/templates/Resources/PractitionerRole.hbs]: Looks good!

File: src/templates/Resources/Specimen.hbs (hunk 6 above)
[the remark this answers is not in the document]
Comment (CaitlinV39) [reply src/templates/Resources/Specimen.hbs]: This is the ORC spreadsheet. Should it be ORC-13?

[the remark this answers is not in the document]
Comment (deepak-ms) [reply src/templates/Resources/Specimen.hbs]: Above link sheet has following :

13 | OBR-13 | Relevant Clinical Information | CWE | 0 | -1 |   |   |   | DiagnosticReport.specimen[1](Specimen.receivedTime) |

**Pull request merged**